crates/revm/src/context.rs

```diff
@@ -154,10 +156,19 @@ impl<EXT, DB: Database> Host for Context<EXT, DB> {
     }
 
     fn code_hash(&mut self, address: Address) -> Option<(B256, bool)> {
-        self.evm
+        let code_hash = self
+            .evm
             .code_hash(address)
             .map_err(|e| self.evm.error = Err(e))
-            .ok()
+            .ok();
+
+        if let Some((bytecode, is_cold)) = self.code(address) {
+            if bytecode.is_eof() {
```

Comment:
@rakita What do you think about doing this instead of checking the first byte?

Reply:
This is fine, as only EOF `0xEFXX` is `0xEF00`

Reply:
thanks for the clarification